src/ui/pages/work/partusage/PartUsageDialog.js

```diff
@@ -202,124 +239,231 @@ function PartUsageDialog(props) {
+            try {
+                const partStockResponse = await WSParts.getPartStock(partCode);
+                const partStock = partStockResponse.body.data;
+
+                // If not in the selected store, explicitly reset the part field since the part code will not be valid in that case.
+                if (
+                    !partStock.some(
+                        (stockEntry) => stockEntry.storeCode === storeCode
+                    )
+                ) {
+                    resetFieldWithDesc(FORM.PART, FORM.PART_DESC);
+                    showError(
+                        `Part "${partCode}" is not in the selected store.`
+                    );
+                    return;
+                }
+            } catch (error) {
+                handleError(error);
+            }
+        }
+
+        const partData = await loadPartData(partCode);
+
+        if (partData?.trackByAsset === 'true') {
+            showNotification(`Selected part "${partCode}" is tracked by asset.`);
+            // Bin loading is done later when the user selects an asset.
+
+        } else if (partData?.trackByAsset === 'false') {
+
+            await Promise.all([
+                loadBinList('', partCode),
+                transactionType === returnType
+                    ? loadLotList(transactionType, '', '', partCode, '')
+                    : null,
+            ]);
+        }
+    };
+
+    const loadLotList = async (transactionType, lot, bin, partCode, storeCode) => {
         try {
-            const partData = await loadPartData(partCode);
-            updateFormDataProperty(FORM.PART_DESC, partData.description);
+            const response = await WSWorkorders.getPartUsageLot(
+                transactionType,
+                lot,
+                bin,
+                partCode,
+                storeCode
+            );
+            const lots = response.body.data;
+
+            if (lots.length === 0) {
+                showWarning('No lots found (likely no available quantity).');
+            }
+
+            if (lots.length === 1) {
+                updateFormDataProperty(FORM.LOT, lots[0].code);
+            }
+
+            if (isHidden(tabLayout['lotcode']) && lots.length >= 1) {
+                updateFormDataProperty(FORM.LOT, '*');
+            }
+
+            setLotList(lots);
         } catch (error) {
             handleError(error);
+            setLotList([]);
         }
     };
 
-    const handlePartChange = (value) => {
-        // Clear asset and bin selection
-        updateFormDataProperty(FORM.ASSET, '');
-        updateFormDataProperty(FORM.ASSET_DESC, '');
-        updateFormDataProperty(FORM.BIN, '');
-        setIsTrackedByAsset(false);
-        // Load the bin list and part data
-        loadBinList('', value);
-        loadPartData(value);
-    };
-
     const loadBinList = async (binCode, partCode) => {
-        if (!partCode) return;
+        if (!partCode) {
+            return;
+        }
```

Comment:
The lack of braces in if statements is allowed iff on short one-liners but this is also fine